### PR TITLE
fix(plasma-core): fix IS_REACT_18

### PR DESCRIPTION
**Pull request by Yeti-or**
### Исправлен детект 18 реакта

– раньше ф-ия проверяла наличие `flushSync` но она присутствует уже в 16 реакте. Теперь мы проверяем совпадение по мажорной версии

:baby_chick: Download canary assets:

[colors--canary.380.4322277405.xml](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/colors--canary.380.4322277405.xml)  
[color_metro_ios-swift--canary.380.4322277405.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_ios-swift--canary.380.4322277405.swift)  
[color_metro_kotlin--canary.380.4322277405.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_kotlin--canary.380.4322277405.kt)  
[color_metro_react-native--canary.380.4322277405.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_react-native--canary.380.4322277405.ts)  
[color_sberHealth_ios-swift--canary.380.4322277405.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_ios-swift--canary.380.4322277405.swift)  
[color_sberHealth_kotlin--canary.380.4322277405.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_kotlin--canary.380.4322277405.kt)  
[color_sberHealth_react-native--canary.380.4322277405.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_react-native--canary.380.4322277405.ts)  
[color_sbermarket_ios-swift--canary.380.4322277405.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_ios-swift--canary.380.4322277405.swift)  
[color_sbermarket_kotlin--canary.380.4322277405.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_kotlin--canary.380.4322277405.kt)  
[color_sbermarket_react-native--canary.380.4322277405.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_react-native--canary.380.4322277405.ts)  
[color_sberprime_ios-swift--canary.380.4322277405.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_ios-swift--canary.380.4322277405.swift)  
[color_sberprime_kotlin--canary.380.4322277405.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_kotlin--canary.380.4322277405.kt)  
[color_sberprime_react-native--canary.380.4322277405.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_react-native--canary.380.4322277405.ts)  
[color_selgros_ios-swift--canary.380.4322277405.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_ios-swift--canary.380.4322277405.swift)  
[color_selgros_kotlin--canary.380.4322277405.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_kotlin--canary.380.4322277405.kt)  
[color_selgros_react-native--canary.380.4322277405.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_react-native--canary.380.4322277405.ts)  
[color_smbusiness_ios-swift--canary.380.4322277405.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_ios-swift--canary.380.4322277405.swift)  
[color_smbusiness_kotlin--canary.380.4322277405.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_kotlin--canary.380.4322277405.kt)  
[color_smbusiness_react-native--canary.380.4322277405.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_react-native--canary.380.4322277405.ts)  
[PlasmaTokensColor--canary.380.4322277405.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/PlasmaTokensColor--canary.380.4322277405.swift)  
[shadow_sbermarket_react-native--canary.380.4322277405.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/shadow_sbermarket_react-native--canary.380.4322277405.ts)  
[typo_mage_ios-swift--canary.380.4322277405.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_ios-swift--canary.380.4322277405.swift)  
[typo_mage_kotlin--canary.380.4322277405.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_kotlin--canary.380.4322277405.kt)  
[typo_mage_react-native--canary.380.4322277405.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_react-native--canary.380.4322277405.ts)  
[typo_plasma_ios-swift--canary.380.4322277405.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_ios-swift--canary.380.4322277405.swift)  
[typo_plasma_kotlin--canary.380.4322277405.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_kotlin--canary.380.4322277405.kt)  
[typo_plasma_react-native--canary.380.4322277405.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_react-native--canary.380.4322277405.ts)  
[typo_ruler_ios-swift--canary.380.4322277405.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_ios-swift--canary.380.4322277405.swift)  
[typo_ruler_kotlin--canary.380.4322277405.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_kotlin--canary.380.4322277405.kt)  
[typo_ruler_react-native--canary.380.4322277405.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_react-native--canary.380.4322277405.ts)  
[typo_sage_ios-swift--canary.380.4322277405.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_ios-swift--canary.380.4322277405.swift)  
[typo_sage_kotlin--canary.380.4322277405.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_kotlin--canary.380.4322277405.kt)  
[typo_sage_react-native--canary.380.4322277405.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_react-native--canary.380.4322277405.ts)  
[typo_sbermarket_ios-swift--canary.380.4322277405.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_ios-swift--canary.380.4322277405.swift)  
[typo_sbermarket_kotlin--canary.380.4322277405.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_kotlin--canary.380.4322277405.kt)  
[typo_sbermarket_react-native--canary.380.4322277405.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_react-native--canary.380.4322277405.ts)  
[typo_soulmate_ios-swift--canary.380.4322277405.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_ios-swift--canary.380.4322277405.swift)  
[typo_soulmate_kotlin--canary.380.4322277405.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_kotlin--canary.380.4322277405.kt)  
[typo_soulmate_react-native--canary.380.4322277405.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_react-native--canary.380.4322277405.ts)

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @salutejs/plasma-b2c@1.152.1-canary.380.4322277405.0
  npm install @salutejs/plasma-core@1.100.1-canary.380.4322277405.0
  npm install @salutejs/plasma-hope@0.11.1-canary.380.4322277405.0
  npm install @salutejs/plasma-icons@1.128.1-canary.380.4322277405.0
  npm install @salutejs/plasma-temple@1.139.1-canary.380.4322277405.0
  npm install @salutejs/plasma-ui@1.172.1-canary.380.4322277405.0
  npm install @salutejs/plasma-web@1.177.1-canary.380.4322277405.0
  npm install @salutejs/plasma-sb-utils@0.98.1-canary.380.4322277405.0
  # or 
  yarn add @salutejs/plasma-b2c@1.152.1-canary.380.4322277405.0
  yarn add @salutejs/plasma-core@1.100.1-canary.380.4322277405.0
  yarn add @salutejs/plasma-hope@0.11.1-canary.380.4322277405.0
  yarn add @salutejs/plasma-icons@1.128.1-canary.380.4322277405.0
  yarn add @salutejs/plasma-temple@1.139.1-canary.380.4322277405.0
  yarn add @salutejs/plasma-ui@1.172.1-canary.380.4322277405.0
  yarn add @salutejs/plasma-web@1.177.1-canary.380.4322277405.0
  yarn add @salutejs/plasma-sb-utils@0.98.1-canary.380.4322277405.0
  ```
</details>

### CODE REVIEW

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-380/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-380/

**Comment by neretin-trike** [comment]
добавишь краткое описание пожалуста?

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out comparison report in job artifacts!

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-380/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-380/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-380/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-380/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-380/temple-storybook/

**Review by neretin-trike — APPROVED** [review]

File: packages/plasma-core/src/utils/react.ts [hunk 1]
```diff
@@ -4,7 +4,7 @@ import ReactDOM from 'react-dom';
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
 const { flushSync } = ReactDOM as any;
 
-export const IS_REACT_18 = typeof flushSync === 'function';
+export const IS_REACT_18 = ReactDOM.version.split('.')[0] === '18';
```
Comment (Yakutoc) [inline packages/plasma-core/src/utils/react.ts:7]: Там точно всегда будет строка? ) 

Нужно дополнительно приводить к строке?

**Review by Yakutoc — APPROVED** [review]

**Pull request merged**